tests/docker/install/install_tests.sh

```diff
@@ -106,11 +106,8 @@ function run_instrumented_server_and_client(){
 
 
 # START TESTING ===========================================
-if [[ "$OSTYPE" == "darwin"* ]]; then
-    HOSTNAME=$(scutil --get LocalHostName)
-else
-    HOSTNAME=$(cat /etc/hostname)
-fi
+# Use Python's socket.gethostname() to match what the agent actually uses on all platforms
+HOSTNAME=$(python -c "import socket; print(socket.gethostname())")
```

Comment:
The macos smoke tests stopped being happy so we update this for macos and linux.